crypto/storage/vault.go

```diff
@@ -0,0 +1,118 @@
+package storage
+
+import (
+	"crypto"
+	"fmt"
+	vault "github.com/hashicorp/vault/api"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+)
+
+const privateKeyPath = "nuts-private-keys"
+const kvEnginePath = "kv"
+const keyName = "key"
+
+type logicaler interface {
+	Read(path string) (*vault.Secret, error)
+	Write(path string, data map[string]interface{}) (*vault.Secret, error)
+}
+
+type vaultKVStorage struct {
+	client logicaler
+}
+
+// NewVaultKVStorage creates a new Vault backend using the kv version 1 secret engine: https://www.vaultproject.io/docs/secrets/kv
+// It currently only supports token authentication which should be provided by the token param.
+// If vaultAddr is empty, the VAULT_ADDR environment should be set.
+// If token is empty, the VAULT_TOKEN environment should be is set.
+func NewVaultKVStorage(token string, vaultAddr string) (Storage, error) {
+	client, err := configureVaultClient(token, vaultAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	vaultStorage := vaultKVStorage{client: client.Logical()}
+	err = vaultStorage.checkConnection()
+	return vaultStorage, err
```

Comment:
I'm not a fan of returning errors like this. I'd rather see it like this for example:


```
if err := vaultStorage.checkConnection(); err != nil {
    return nil, err
}

return vaultStorage, nil
```

Otherwise it's hard to see wether the error was actually checked